src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java

```diff
@@ -19,10 +21,17 @@
 public class FlashcardFacade {
     private FlashcardService flashcardService;
     private QuestionTagService questionTagService;
+    private UserService userService;
     private ModelMapper modelMapper;
 
     public FlashcardSaveResponseDto saveFlashcard(FlashcardSaveDto flashcardSaveDto) {
         Flashcard flashcard = modelMapper.map(flashcardSaveDto, Flashcard.class);
+        if(!flashcardSaveDto.getUserId().isEmpty()) {
+            User user = userService.findById(Long.parseLong(flashcardSaveDto.getUserId()));
```

Comment:
Is there any reason to keep `flashcardSaveDto.userId` of type `String`? Maybe let's change it to `Long` and then you don't need to cast it here.